Apps/W1/HybridGP/app/src/Migration/Customers/GPCustomerMigrator.codeunit.al

```diff
@@ -382,17 +396,111 @@ codeunit 4018 "GP Customer Migrator"
                 GPMigrationWarnings.InsertWarning(MigrationLogAreaTxt, WarningContext, StrSubstNo(PhoneNumberContainsLettersMsg, MigrationGPCustomer.FAX));
     end;
 
-    local procedure MigrateCustomerAddresses(MigrationGPCustomer: Record "GP Customer")
+    local procedure MigrateCustomerAddresses(var MigrationGPCustomer: Record "GP Customer"; var Sender: Codeunit "Customer Data Migration Facade")
     var
         GPCustomerAddress: Record "GP Customer Address";
+        CustomerNo: Code[20];
     begin
-        GPCustomerAddress.SetRange(CUSTNMBR, MigrationGPCustomer.CUSTNMBR);
+        CustomerNo := CopyStr(MigrationGPCustomer.CUSTNMBR, 1, MaxStrLen(CustomerNo));
+        if not Sender.DoesCustomerExist(CustomerNo) then
+            exit;
+
+        GPCustomerAddress.SetRange(CUSTNMBR, CustomerNo);
         if GPCustomerAddress.FindSet() then
             repeat
                 GPCustomerAddress.MoveStagingData();
             until GPCustomerAddress.Next() = 0;
     end;
 
+    local procedure CreateRecurringSalesLines(var GPCustomer: Record "GP Customer"; var Sender: Codeunit "Customer Data Migration Facade")
+    var
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        GPRM00201: Record "GP RM00201";
+        GPPostingAccounts: Record "GP Posting Accounts";
+        CustomerNo: Code[20];
+    begin
+        if not GPCompanyAdditionalSettings.GetGLModuleEnabled() then
+            exit;
+
+        CustomerNo := CopyStr(GPCustomer.CUSTNMBR, 1, MaxStrLen(CustomerNo));
```

Comment:
Can we add a setting to disable this code or an event to skip the code if possible?
Maybe not everyone wants to move this data.

Reply:
Added config setting